test/test_benchmark.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2024 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Benchmark tests for ops and ops-scenario.
+
+Optimising performance is not a current goal with either ops or
+ops-scenario - any gains are unlikely to be significant compared with ones
+from Juju or the charm and its workload. However, we do want to ensure that
+we do not unknowingly regress in performance.
+
+This module contains a small set of tests that cover core functionality,
+to be used for performance benchmarking.
+"""
+
+import dataclasses
+import pathlib
+import sys
+
+import ops
+from ops import testing
+
+sys.path.append(str(pathlib.Path(__file__).parent / 'charms' / 'test_benchmark' / 'src'))
+
+from bcharm import BenchmarkCharm
+
+
+def test_context_explicit_meta(benchmark):
```

Comment:
Magical pytest import strikes again! I couldn't for the life of me see where this came from at first. All I want for Christmas is for pytest to start being explicit and acting like regular Python! :-)

Reply:
> Magical pytest import strikes again! I couldn't for the life of me see where this came from at first. All I want for Christmas is for pytest to start being explicit and acting like regular Python! :-)

:) I'm not a fan of this either. I could add a comment explaining that the fixture is automatically available via pytest-benchmark above the first test, or in the module docstring?

Reply:
That would be helpful, thanks.

Reply:
I think above the first test would be more visible.